### PR TITLE
Don't omit local parameters named api-version

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
The logic for embedding client property api-version as a constant query
parameter value was inadvertently removing api-version when specified as
a method parameter leading to incorrect codegen.

Pin test server version and update tests.

### CODE REVIEW

File: src/Model/ParameterGo.cs [hunk 1]
```diff
@@ -61,11 +55,9 @@ public string GetParameterName()
             return retval;
         }
 
-        public override bool IsClientProperty => base.IsClientProperty == true && !IsAPIVersion;
-
         public virtual bool IsAPIVersion => SerializedName.IsApiVersion();
 
-        public virtual bool IsMethodArgument => !IsClientProperty && !IsAPIVersion && !IsConstant;
```
Comment (jhendrixMSFT) [inline src/Model/ParameterGo.cs]: This is the meat of the fix, the rest is removing needless clutter.

**Comment by jhendrixMSFT** [comment]
CC @catalinaperalta

**Comment by fearthecowboy** [comment]
# 🤖 AutoRest automatic publish job 🤖
## success (version: 2.1.139)

**Comment by jhendrixMSFT** [comment]
@ArcturusZhang could you try out this new build and if it looks ok update the latest branch?  I built a handful of services and it looked ok.

**Comment by paulomarquesc** [comment]
Thanks guys for the quick repsone on this.

**Comment by ArcturusZhang** [comment]
> @ArcturusZhang could you try out this new build and if it looks ok update the latest branch? I built a handful of services and it looked ok.

Sure

**Pull request merged**